doc/zk-integration.md

```diff
@@ -0,0 +1,264 @@
+# Integration with ZK rollups
+
+## Background
+
+> For more context and terminology about the architecture of Espresso, please read [README.md](../README.md).
+
+At a high level ZK rollups produce blocks and periodically settle their new state on layer 1 (e.g. Ethereum), after
+applying all the transactions of such blocks to their current state. The time between two consecutive updates can vary,
+yet it is high enough in order to amortize gas costs for the rollup and thus offer low fees to end users. ZK rollups
+have the option to build their blocks by reading from HotShot, the Espresso consensus protocol, that produces blocks
+containing transactions from multiple rollups. In this setting each rollup is identified by an identifier called
+_namespace_. Moreover, HotShot generates an authenticated piece of data, the _finality gadget_, that guarantees that a
+specific state of the Espresso ledger will not be reverted. ZK rollups can leverage this finality gadget and a scheme to
+filter transactions by namespace in order to prove their source of transactions is the Espresso ledger. Note that in
```

Comment:
nit: prefer "notarized by Espresso finality gadget" instead of referring to a ledger.

Reply:
Done in a041bd44.